.azure-pipelines/cd-build-deploy-typescript.yml

```
@@ -0,0 +1,164 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Build, package, and deploy TypeScript packages for @microsoft/agents-m365copilot-core and @microsoft/agents-m365copilot-beta
+
+name: $(BuildDefinitionName)_$(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)
+
+trigger:
+  branches:
+    include:
+      - main
+  paths:
+    include:
+      - 'typescript/packages/agents-m365copilot-beta/generated/**'
+      - 'typescript/packages/agents-m365copilot-core/src/**'
+  tags:
+    include:
+      - '*microsoft/agents-m365copilot-beta'
+      - '*microsoft/agents-m365copilot-core'
+
+pr: none
+
+variables:
+  package_name_core: 'agents-m365copilot-core'
+  package_name_beta: 'agents-m365copilot-beta'
+  package_dir_core: 'packages/agents-m365copilot-core'
+  package_dir_beta: 'packages/agents-m365copilot-beta'
+
+resources:
+  repositories:
+  - repository: 1ESPipelineTemplates
+    type: git
+    name: 1ESPipelineTemplates/1ESPipelineTemplates
+    ref: refs/tags/release
+
+extends:
+  template: v1/1ES.Official.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    pool:
+      name: Azure-Pipelines-1ESPT-ExDShared
+      vmImage: windows-latest
+    stages:
+
+    - stage: build
+      displayName: 'Build and Package TypeScript Packages'
+      # This condition allows the build to run for both tagged builds and manual builds.
+      condition: or(contains(variables['Build.SourceBranch'], '@microsoft/agents-m365copilot'), eq(variables['Build.Reason'], 'Manual'))
+      jobs:
+        - job: build_typescript
+          displayName: 'Build, Test, and Package Core and Beta'
+          pool:
+            name: Azure-Pipelines-1ESPT-ExDShared
+            vmImage: ubuntu-latest
+
+          steps:
+            - task: NodeTool@0
+              displayName: 'Install Node.js'
+              inputs:
+                versionSpec: '20.x'
```

```suggestion
                version: '22.x'
```
https://endoflife.date/nodejs